### PR TITLE
Rustify utils::is_http_url

### PR DESCRIPTION
**Pull request by rnestler**
See https://github.com/newsboat/newsboat/issues/334

### CODE REVIEW

File: test/utils.cpp [hunk 1]
```diff
@@ -348,6 +348,18 @@ TEST_CASE("strwidth()", "[utils]")
...
+TEST_CASE("is_http_url()", "[utils]")
+{
+	REQUIRE(utils::is_http_url("http://foo.bar"));
+	REQUIRE(utils::is_http_url("https://foo.bar"));
+	REQUIRE(utils::is_http_url("http://"));
+	REQUIRE(utils::is_http_url("https://"));
+
+	REQUIRE_FALSE(utils::is_http_url("htt://foo.bar"));
+	REQUIRE_FALSE(utils::is_http_url("http:/"));
+	REQUIRE_FALSE(utils::is_http_url("foo://bar"));
```
Comment (rnestler) [inline test/utils.cpp]: I'm not sure if it is really worth to have that many tests for such a simple function or if there would be better test cases, but here they are :wink:

Comment (Minoru) [reply test/utils.cpp]: Yeah, 6 tests are plenty. Thanks! Please port them to Rust, too. This way, when we remove the C++ "shell" of the function, the actual code is still covered by tests.

**Comment by Minoru** [comment]
Looks great! I'll merge when you copy the tests to Rust.

File: test/utils.cpp (hunk 1 above)
Comment (rnestler) [reply test/utils.cpp]: Ah that makes sense. Done.

**Comment by Minoru** [comment]
Cool, thanks a lot @rnestler!

**Pull request merged**